django/contrib/admin/options.py

```diff
@@ -561,14 +576,33 @@ def wrapper(*args, **kwargs):
         urlpatterns = [
             url(r'^$', wrap(self.changelist_view), name='%s_%s_changelist' % info),
             url(r'^add/$', wrap(self.add_view), name='%s_%s_add' % info),
+            url(r'^autocomplete/(?P<field_name>.+)/$',
+                wrap(AutocompleteJsonView.as_view(
+                    admin_site=self.admin_site,
+                    source_model=self.model,
+                    source_model_admin=self,
+                )), name='%s_%s_autocomplete' % info),
             url(r'^(.+)/history/$', wrap(self.history_view), name='%s_%s_history' % info),
             url(r'^(.+)/delete/$', wrap(self.delete_view), name='%s_%s_delete' % info),
             url(r'^(.+)/change/$', wrap(self.change_view), name='%s_%s_change' % info),
+        ]
+        for inline in self.get_inline_instances(request=None):
```

Comment:
Will do, fwiw I think 
```
url(r'^autocomplete_inline/%s/%s/(?P<field_name>.+)/$' % inline_info,
```
is not enough, two models could use the same inlines with different configs.